File: Data Structures & Algorithms/trapping-rain-water/submission-1.cpp
class Solution {
public:
    int trap(vector<int>& height) {
        int left = 0;
        int right = height.size() - 1;
        int maxLeft = height[left];
        int maxRight = height[right];
        int total = 0;
        while (left < right) {
            if (maxLeft < maxRight) {
                left++;
                total += max(0, min(maxLeft, maxRight) - height[left]);
                maxLeft = max(maxLeft, height[left]);
            }
            else {
                right--;
                total += max(0, min(maxLeft, maxRight) - height[right]);
                maxRight = max(maxRight, height[right]);
            }
        }
        return total;
    }
};

File: Data Structures & Algorithms/longest-palindromic-substring/submission-1.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        longest = ""
        for i in range(len(s)):
            for j in [0, 1]:
                left, right = i, i + j
                while left > -1 and right < len(s):
                    if s[left] == s[right]:
                        if right - left + 1 > len(longest):
                            longest = s[left: right+1]
                        left -= 1
                        right += 1
                    else:
                        break
        return longest